cluster/gce/gci/flexvolume_node_setup.sh

```diff
@@ -99,31 +99,31 @@ generate_chroot_wrapper() {
     exit 1
   fi
 
-  for driver_dir in ${MOUNTER_PATH}/flexvolume/*; do
+  for driver_dir in "${MOUNTER_PATH}/flexvolume"/*; do
     if [ -d "$driver_dir" ]; then
 
-      filecount=$(ls -1 $driver_dir | wc -l)
-      if [ $filecount -gt 1 ]; then
+      filecount=$(cd "$driver_dir"; find . -mindepth 1 -maxdepth 1 -print0 | xargs -0 -n1 basename | grep -c -v "^\.")
+      if [ "$filecount" -gt 1 ]; then
         echo "ERROR: Expected 1 file in the FlexVolume directory but found $filecount."
         exit 1
       fi
 
-      driver_file=$( ls $driver_dir | head -n 1 )
+      driver_file=$(cd "$driver_dir"; find . -mindepth 1 -maxdepth 1 -print0 | xargs -0 -n1 basename | grep -v "^\." | head -n 1)
```

Comment:
we should prefer cut to grep -v since we are removing a known fixed length prefix from all entries.

https://github.com/kubernetes/kubernetes/blob/8c6c94bad2ef70c98d38fc5ebb3ad28f173c683b/hack/lib/util.sh#L541-L544

Reply:
`find` reveals files starting with `.`(ex. `.aaa`). But `ls` without `-a` does not. So, `grep -v` in this line excepts the files.

Reply:
I see now, we already pipe through basename which drops the `./foo/`, but do we actually need to ignore these?

Reply:
If find outputs the following result, I think `grep -v` is needed to exclude `.ddd`.
```
$ cd find_test; find . -mindepth 1 -print0 | xargs -0 -n1 basename
aaa
.ddd
bbb
ccc
```
Should we not consider these case?

Reply:
I'm not sure, I guess it depends on the data. Let's handle it for now but maybe leave a TODO about it. 

also cc @msau42 ... does it make sense to ignore drivers starting with `.` ?

Reply:
directories that start with `.` are reserved for the driver installation mechanism (to make sure driver installs are atomic). See [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/flexvolume-deployment.md#recommended-driver-deployment-method) for details

Reply:
like I said up top for both of these now-find-based-implementations, I feel like the original ls-based implementations are fine; it's not clear to me what value we're getting here by adding all of this extra complexity to work around corner cases that haven't crept up thus far

Reply:
> directories that start with . are reserved for the driver installation mechanism (to make sure driver installs are atomic). See here for details

directories sure, files though?

Reply:
> like I said up top for both of these now-find-based-implementations, I feel like the original ls-based implementations are fine; it's not clear to me what value we're getting here by adding all of this extra complexity to work around corner cases that haven't crept up thus far

the `ls` implementation was already not-correct I suspect. in neither of these are we currently only finding files, despite the variable name.

Reply:
Thank you for many comments.
I've checked flexvolume behavior.

When I placed `kubelet-plugins/volume/exec/test~.aaa/.aaa`, which is dummy driver, `.aaa` can work.
From the result, I think it should not except files begin with ".".

I will modify as the below.
1. (line:105) `filecount=$(cd "$driver_dir"; find . -mindepth 1 -maxdepth 1 -print0 | xargs -0 -n1 basename | wc -l)`
1. (line:111) `driver_file=$(cd "$driver_dir"; find . -mindepth 1 -maxdepth 1 -print0 | xargs -0 -n1 basename | head -n 1)`

Reply:
SGTM, thanks for being thorough!

Reply:
Thanks, I'm done.